src/uikit/Button.tsx

```diff
@@ -0,0 +1,81 @@
+import type {
+  ButtonProps as BaseButtonProps,
+  ChakraComponent,
+  CSSWithMultiValues,
+} from '@chakra-ui/react';
+import { Button as BaseButton, useStyleConfig } from '@chakra-ui/react';
+import type { ReactNode } from 'react';
+import { useEffect, useRef, useState } from 'react';
+
+const maxDegree = 90;
+
+type ButtonComponent = ChakraComponent<'button', ButtonProps>;
+
+interface ButtonProps extends BaseButtonProps {
+  children: ReactNode;
+  isAnimated?: boolean;
+}
+
+function Button({ children, isAnimated, ...rest }: ButtonProps) {
```

Comment:
How about change the name to more specific like "AnimationButton". So we can know the button component by just seeing the component name "ohh this is button from chakra or custom"

Reply:
I don't agree because the button itself is not intended for animation and is only for a basic "mocha" button. How about we use an approach like inheriting/merging the base mocha button component?

Reply:
ohh my bad. `<Button/>` that used in other component already use this component yes. so I'm okay with this naming component